modules/network-observability-working-with-connections.adoc

```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// network_observability/observing-network-traffic.adoc
+
+:_content-type: PROCEDURE
+[id="network-observability-working-with-conversations_{context}"]
+= Working with conversation tracking
+As an administrator, you can group network flows that are part of the same conversation. A conversation is defined as a grouping of peers that are identified by their IP addresses, ports, and protocols resulting in a unique *Conversation ID*. Events that are associated with the conversation are queryable in the web console. These events are represented in the web console as follows:
+
+** *Conversation start* happens when an unknown connection is starting or TCP flag intercepted
+** *Heartbeat* occurs at every specified interval defined in flow collector spec.processor.connectionUpdateInterval while the connection is up.
+** *Conversation end* occurs when the `FlowCollector` `spec.processor.connectionEndTimeout` is reached or TCP flag intercepted.
```

Comment:
I meant here: shouldnt conversation be connection in all the above places?

Reply:
Conversation is the UI wording of connection tracking. The configurations and data behind mention `connection` everywhere.

Also our CRD keep the technical wording:
```
	// outputRecordTypes defines the desired record types to generate. Possible values are "FLOWS" (default) to export
	// flowLogs, or "ALL" to generate both flowLogs and newConnection, heartbeat, endConnection events
	// +kubebuilder:validation:Optional
	// +kubebuilder:validation:Enum:="FLOWS";"ALL"
	// +kubebuilder:default:=FLOWS
	OutputRecordTypes *string `json:"outputRecordTypes,omitempty"`

	//+kubebuilder:default:="30s"
	// +optional
	// connection heartbeat interval is the duration of time to wait between heartbeat reports of a connection
	ConnectionHeartbeatInterval *metav1.Duration `json:"connectionHeartbeatInterval,omitempty"`

	//+kubebuilder:default:="10s"
	// +optional
	// connection end timeout is the duration of time to wait from the last flow log to end a connection
	ConnectionEndTimeout *metav1.Duration `json:"connectionEndTimeout,omitempty"`
```

I feel we are good as is if we keep it clear for the user in the `Query options` doc section. WDYT ?

Reply:
renamed CRD fields for consistency:
- `outputRecordTypes` to `logTypes`
- `connectionEndTimeout` to `conversationEndTimeout`
- `connectionHeartbeatInterval` to `conversationHeartbeatInterval`

https://issues.redhat.com/browse/NETOBSERV-350
https://github.com/netobserv/network-observability-operator/pull/303